google/ggrc-core
# [5OBJ] GGRC-8532 Disable create, update, delete for 5OBJ, their Custom Attributes and Roles

# Dependencies

_**This PR should be merged into [GGRC-7200](https://github.com/google/ggrc-core/tree/GGRC-7200) branch which is the common one for all 5OBJ. PRs**_

This PR is `on hold` until the dependencies are merged:

- [x] #10615 (Move Contracts & Policies out into separate tables)
- [x] #10631 (Extend 5OBJ with Q-side fields)
- [x] #10602 (Clean HTML from 5OBJ attributes and comments)
- [x] #10641 (Move 5OBJ's comments into external)

# Issue description

Create, update and delete actions via REST API should be disabled for 5OBJ from C side.

# Steps to test the changes

1. Run the necessary migrations;
2. Make sure that create, update, and delete operations are forbidden for 5OBJ via REST API for **internal** users.
3. Make sure that create, update, and delete operations are allowed for 5OBJ via REST API for **external** users.

# Solution description

In scope of this PR the following changes were made:

* Add separate classes for `Contract` and `Policy`, and remove corresponding classes derived from `Directive` one.

* Extend all 5OBJ classes with new following parents: `Synchronizable ` (forbids CRUD operations for objects and adds "external_id", "external_slug" fields), `WithExternalCreatedBy` (adds "Created By" field), and `ExternalCommentable` (allows creation of external comments).

  Remove `Commentable` (internal comments should not be supported) and `Reviewable` (review should not be supported) from parent classes.

* Adjust already existing integration tests (fix or remove).

* Add new integration tests to cover CRUD for 5OBJ from both internal and external users point of view.

# Sanity checklist




# PR Review checklist
